CONTRIBUTING.md

```diff
@@ -0,0 +1,87 @@
+# Contributing Guidelines
+
+[![](https://img.shields.io/badge/made%20by-Protocol%20Labs-blue.svg?style=flat-square)](http://ipn.io)
+[![](https://img.shields.io/badge/project-libp2p-blue.svg?style=flat-square)](https://libp2p.io/)
+
+Welcome to the rust-libp2p contribution guide! We appreciate your interest in improving our library.
+
+## Table of Contents
+
+- [Contributing Guidelines](#contributing-guidelines)
+  - [Table of Contents](#table-of-contents)
+  - [Community Guidelines](#community-guidelines)
+  - [Looking for ways to contribute?](#looking-for-ways-to-contribute)
+  - [Code Contribution Guidelines](#code-contribution-guidelines)
+    - [Discuss big changes as Issues first](#discuss-big-changes-as-issues-first)
+    - [Welcoming Pull Requests](#welcoming-pull-requests)
+    - [Code and Conventions](#code-and-conventions)
+      - [Commits](#commits)
+      - [Pull Requests](#pull-requests)
+      - [Tests](#tests)
+      - [Documentation](#documentation)
+      - [Writing Changelog Entries](#writing-changelog-entries)
+      - [Bumping Versions](#bumping-versions)
+      - [Mergify and the Send-it Label](#mergify-and-the-send-it-label)
+  - [Credits](#credits)
+
+## Community Guidelines
+
+At rust-libp2p, we value a friendly and inclusive community. Harassment and abusive behavior are strictly prohibited. Please be respectful and kind to fellow contributors.
+
+## Looking for ways to contribute?
+
+There are several ways you can contribute to rust-libp2p:
+- Start contributing immediately via the opened [issues](https://github.com/libp2p/rust-libp2p/issues).
+- Reporting issues, bugs, mistakes, or inconsistencies.
+- Submitting a feature request.
+- Opening a new pull request: primary means of making concrete changes to the code.
+
+## Code Contribution Guidelines
+
+### Discuss big changes as Issues first
+
+Before starting to code significant improvements, it's important to document them as GitHub issues in order to get feedback.
+
+### Welcoming Pull Requests
+
+Pull requests are the primary means of making concrete changes to the code, documentation, and dependencies of rust-libp2p. Even small pull requests, such as fixing a typo, are highly appreciated.
+
+### Code and Conventions
+
+#### Commits
+
+It is highly recommended to organize your changes into logically grouped commits. Keeping each commit focused on a specific aspect of the changes makes it easier for reviewers to understand and evaluate the modifications.
+
+Commit messages must start with a short subject line, followed by an optional, more detailed explanatory text which is separated from the summary by an empty line.
```

Comment:
We don't care much about the commits within a PR actually because we squash merge.